### PR TITLE
AB#80602 Deactivate option to see datasource in text widget and summary card widget if using ref data

### PR DESCRIPTION
**Pull request by guisoares1**
# Description
When using ref data, the option `Show link to data source` should not be visible in text widget and summary card widget.

# Error found and fixed
- The option `Show link to data source` in text widget was not being saved or loaded. ([commit](https://github.com/ReliefApplications/ems-frontend/pull/2120/commits/ef09bdf6624e8833366c30c83f81dc2e17abd345))

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/80602)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] A data reference was added to each of the widgets and the behavior was observed. Furthermore, I made sure that the logic that already existed continued to work.

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/55603259/096d4879-60c0-4c6d-86f2-9f072751c7a6

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Review by RenzoPrats — APPROVED** [review]
Working fine, looks good to me!

File: libs/shared/src/lib/components/widgets/summary-card-settings/display-tab/display-tab.component.html [hunk 1]
```diff
@@ -15,7 +15,10 @@ <h2>
       </div>
       <ui-divider class="my-1"></ui-divider>
       <div class="flex flex-col gap-2">
-        <ui-toggle formControlName="showDataSourceLink">
+        <ui-toggle
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widgets/summary-card-settings/display-tab/display-tab.component.html:18]: @RenzoPrats @guisoares1 
simpler solution to display or not settings
and more dynamic

File: libs/shared/src/lib/components/widgets/summary-card/summary-card.component.ts [hunk 2]
```diff
@@ -181,6 +181,16 @@ export class SummaryCardComponent
     return !isNil(this.settings.card?.referenceData);
   }
 
+  /** @returns should show data source button */
+  get showDataSourceButton() {
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widgets/summary-card/summary-card.component.ts:185]: @RenzoPrats @guisoares1 
also more dynamic to do that

**Comment by AntoineRelief** [comment]
@RenzoPrats @guisoares1 
in most cases, the logic was working, but you should prefer dynamic solutions ( especially for settings ) like the one I put in the last commit 👍

**Pull request merged**